Orange/preprocess/impute.py

```diff
@@ -107,11 +107,14 @@ def __call__(self, data, variable):
         variable = data.domain[variable]
         domain = domain_with_class_var(data.domain, variable)
         data = data.from_table(domain, data)
-        model = self.learner(data)
-        assert model.domain.class_var == variable
-        return variable.copy(
-            compute_value=ReplaceUnknownsModel(variable, model))
 
+        if self.learner.check_learner_adequacy(domain):
+            model = self.learner(data)
+            assert model.domain.class_var == variable
+            return variable.copy(
+                compute_value=ReplaceUnknownsModel(variable, model))
+        else:
+            return variable.copy()
```

Comment:
When learner on input is not compatible with the variable, the variable does not get imputed.

I would rather move the check to the imputation widget. When the learner on input does not support variable type, a Model imputer with default_learner could be constructed.